torch/testing/__init__.py

```diff
@@ -27,35 +155,15 @@ def assert_allclose(actual, expected, rtol=None, atol=None, equal_nan=True, msg=
             raise ValueError("rtol and atol must both be specified or both be unspecified")
         rtol, atol = _get_default_tolerance(actual, expected)
 
-    close = torch.isclose(actual, expected, rtol, atol, equal_nan)
-    if close.all():
+    result, debug_msg = _compare_tensors_internal(actual, expected,
```

Comment:
Is this potentially slower? I believe torch.isclose is implemented as a single c++ kernel.

Also does this change anything BC-breaking? Users do rely on torch.testing.assert_allclose in their own unit tests.

Reply:
It is potentially slower but the current behavior is incorrect and most of the slowdown only comes when the test fails, which is probably not a scenario we need to optimize for.

It *could* be BC-breaking but I wouldn't expect it to be unless people were testing complex tensors, which is very unlikely.

Reply:
I do see weird behavior on master for isclose, so I guess we can't break this. For some reason I thought there was an implementation in isclose using angle:
```
>>> torch.isclose(torch.tensor(1+1j), torch.tensor(1+1j))
tensor(False)
```
cc @anjali411 FYI in case there's any concern.

I was thinking more about nan masks and such that might be computed differently here than there, but looking again I don't see anything outside complex that isn't short-circuited by an isclose anyway.

Reply:
Ah. I fixed the nan mask behavior per prior comment. That example you're showing is a problem, though. We even have a test for torch.isclose that should be catching issues like that!

Reply:
Wait, on master this:

```
print(torch.isclose(torch.tensor(1+1j), torch.tensor(1+1j)))
```
prints tensor(True) for me. The fix for isclose went in a couple weeks ago. Do you need to update?

Reply:
yes that works in another install, I had a prompt open with an older checkout

Reply:
Whew!